scripts/globals/beastmentreasure.lua

```diff
@@ -0,0 +1,394 @@
+    [125] = convertToWeighted(addLoot(sharedLoot.seeds, zoneData[125].loot.racial))
+}
+local w_coins       = convertToWeighted(sharedLoot.coins)
+local w_sealsunique =
+{
+    [123] = convertToWeighted(addLoot(sharedLoot.seals, zoneData[123].loot.unique)),
+    [124] = convertToWeighted(addLoot(sharedLoot.seals, zoneData[124].loot.unique)),
+    [125] = convertToWeighted(addLoot(sharedLoot.seals, zoneData[125].loot.unique)),
+}
+
+local function weightedRandomSelect(w_loot)
+    -- Takes a weighted loot table and returns a single random result
+    return w_loot[math.random(#w_loot)]
+end
+
+local function getZoneID(entity)
+    return entity:getZone():getID()
+end
+
+local function getZoneData(entity)
+    return zoneData[getZoneID(entity)]
+end
+
+local function getRandomQm(digsites)
+    --[[ We specify digsite indices in zoneData that correspond to their qm number.
+    For example, zoneData[123].digsites[10] refers to qm10 in zone 123 (Yuhtunga Jungle).
+    This has a few advantages, but it also makes the table nonconsecutive, which means Lua
+    can't find its length without iteration. ]]--
+    local qms = {}
+    for qm in pairs(digsites) do
+        table.insert(qms, qm)
+    end
+    return qms[math.random(#qms)]
+end
+
+local function getAssignedDigSite(player, zd)
+    return zd.digsites[player:getVar(zd.qmvar)]
+end
+
+local function startMapMarkerEvent(eventid,player,zd)
+    --[[ Map marker event notes:
+        Event 101 creates the "Treasure" map marker using coordinate args
+        Event 103 shows the "Treasure" map marker using coordinate args
+        Event 105, played on excavation, deletes the "Treasure" map marker
+
+        Parameters for events 101/103 are as follows:
+            1: zone id
+            2: unknown, seems static
+            3: (x position of qm) * 1000
+            4: (y position of qm) * 1000
+            5: unknown, dynamic
+            6: unknown, dynamic
+            7: unknown, seems static
+            8: unknown*, seems static (possibly z position of qm + 95)
+
+        Example: `player:startEvent(101, 123, 0, 369795, 201805, 0, 0, 0, 4095)`
+            ...plays event 101 in Yuhtunga Jungle (zone 123), which creates a map marker
+            corresponding to the QM at position (369.795, 4.0, 201.805)
+            Full args from a retail capture for this point were:
+                101, 123, 32, 369795, 201805, 425920, 760, 529191, 4095
+                103, 123, 32, 369795, 201805, 308060, 1694, 529191, 4095 ]]--
+    local zoneid = getZoneID(player)
+    local digsite = getAssignedDigSite(player, zd)
+
+    player:startEvent(eventid, zoneid, 0, digsite.x * 1000, digsite.y * 1000)
+end
+
+dsp.beastmentreasure.handleNpcOnTrigger = function(player)
+    local zd = getZoneData(player)
+    local status = player:getVar(zd.statusvar)
+    local hasMap = player:hasKeyItem(zd.mapid)
+
+    if not hasMap then
+        player:startEvent(102) -- Peddlestox lectures you for not having a map
```

Comment:
I greatly prefer an explanatory statement about what an event is, as it makes the _why_ in a script's logic more readily apparent, making a quest easier to understand and debug.

When you're looking at a quest for the first time you're not going to have all the event IDs memorized, so knowing the gist of an event helps you understand the context of why the scripter is triggering it when they are (and therefore, see if they have made a mistake or not).

Reply:
Personally I like it when event ids have a quick summary in the comments so that you don't have to play back every cutscene to understand what they do, but these can be dropped if we want.

Reply:
personally I preferred the header of the NPC to list them but thats only been done in a few NPCs in the distant past before work on an extractor tool was started. I think once that is released this won't matter.

Reply:
We could move these to the NPC headers, but they'd be identical across all 3 Peddlestox scripts. Seems to make sense to have one copy here where the logic is actually handled.

Reply:
replace "header of the npc" with "header of the script calling them" then. back when we actually kept the commented we didn't have globals handling events other than bcnm /shrug.

But in the (hopefully near) future we'll be able to just open up a list for them and see what they are.

Reply:
If they're in the header of the NPC/script, aren't the events still being summarized by comments?

(Also scrolling back-and-forth to look in the header for the event being referenced further down the file.)

Reply:
doesn't really matter where they are or if they exist at this point. even commenting isn't a huge debugging tool for whats already implemented, more a tool for "this is what I am adding"/"these are cs involving this NPC that have not been implemented yet"

I was only making a passing comment about how I used to like them when we even did that: every cs the NPC had was all in one spot.